src/components/Header.tsx

```diff
@@ -0,0 +1,122 @@
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { DispatchContext, StateContext } from '../management/TodoContext';
+import { USER_ID, createTodos } from '../api/todos';
+
+export const Header: React.FC = () => {
+  const dispatch = useContext(DispatchContext);
+  const { todos, isLoading } = useContext(StateContext);
+  const userId = USER_ID;
+  const [title, setTitle] = useState('');
+
+  const completedAll = todos.every(todo => todo.completed);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (inputRef.current && !title && !isLoading) {
+      inputRef.current.focus();
+    }
+  }, [title, isLoading]);
+
+  const hendleAddedTodo = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (inputRef.current) {
+      inputRef.current.disabled = true;
+    }
+
+    const data = {
+      title: title.trim(),
+      userId,
+      completed: false,
+    };
+
+    if (data.title.trim()) {
+      dispatch({
+        type: 'tempTodo',
+        payload: {
+          createdAt: '',
+          updatedAt: '',
+          id: 0,
+          userId,
+          title: title.trim(),
+          completed: false,
+        },
+      });
+
+      createTodos(data)
+        .then(newTodo => {
+          dispatch({
+            type: 'addTodo',
+            payload: newTodo,
+          });
+          setTitle('');
+        })
+        .catch(() => {
+          dispatch({
+            type: 'errorMessage',
+            payload: 'Unable to add a todo',
+          });
+        })
+        .finally(() => {
+          if (inputRef.current) {
+            inputRef.current.disabled = false;
+            inputRef.current.focus();
+          }
+
+          dispatch({ type: 'tempTodo', payload: null });
+        });
+    } else {
+      dispatch({
+        type: 'errorMessage',
+        payload: 'Title should not be empty',
+      });
+
+      if (inputRef.current) {
+        inputRef.current.disabled = false;
+        inputRef.current.focus();
+      }
+    }
+  };
+
+  const hendleChangeStatusAll = () => {
+    dispatch({
+      type: 'changeStatusAll',
+      payload: !completedAll,
+    });
+  };
+
+  return (
+    <header className="todoapp__header">
+      {todos.length > 0 && (
```

Comment:
```suggestion
      {!!todos.length && (
```

Reply:
If I change "todos.length > 0 &&" to "!!todos.length &&", then one test does not pass, and if I do not touch, then all tests pass.
![Screenshot 2024-02-11 225304](https://github.com/mate-academy/react_todo-app-add-and-delete/assets/122085233/223d309b-ecca-4af7-8dea-f49305f50f23)